qiskit/_quantumcircuit.py

```diff
@@ -54,6 +54,11 @@ def __init__(self, *regs, name=None):
         self._increment_instances()
         if name is None:
             name = self.cls_prefix() + str(self.cls_instances())
+
+        if not isinstance(name, str):
```

Comment:
Can you update the docstring of `__init__` with the information about `name`? `name (str or None): ....`

Reply:
Fixed in 4db07e4